src/core/MOM_CoriolisAdv.F90

```diff
@@ -1050,6 +1422,449 @@ subroutine gradKE(u, v, h, KE, KEx, KEy, k, OBC, G, GV, US, CS)
+  real :: tau                 ! Temporary variables [nondim]
+  real :: w0, w1, w2, w3      ! Weights [nondim]
+  real :: s                   ! Temporary variables [nondim]
+
+  if (u>0.) then
+    call weno_seven_reconstruction_0(q8(4:7), c0) ! Reconstruction in the fourth upwind stencil
+    call weno_seven_reconstruction_1(q8(3:6), c1) ! Reconstruction in the third upwind stencil
+    call weno_seven_reconstruction_2(q8(2:5), c2) ! Reconstruction in the second upwind stencil
+    call weno_seven_reconstruction_3(q8(1:4), c3) ! Reconstruction in the first upwind stencil
+
+    call weno_seven_reconstruction_0(h8(4:7), d0) ! Reconstruction in the fourth upwind stencil
+    call weno_seven_reconstruction_1(h8(3:6), d1) ! Reconstruction in the third upwind stencil
+    call weno_seven_reconstruction_2(h8(2:5), d2) ! Reconstruction in the second upwind stencil
+    call weno_seven_reconstruction_3(h8(1:4), d3) ! Reconstruction in the first upwind stencil
+    if (velocity_smoothing) then
+      call weno_seven_weight_0(u8(4:7), b0)       ! Smoothness indicator of the fourth upwind stencil
+      call weno_seven_weight_1(u8(3:6), b1)       ! Smoothness indicator of the third upwind stencil
+      call weno_seven_weight_2(u8(2:5), b2)       ! Smoothness indicator of the second upwind stencil
+      call weno_seven_weight_3(u8(1:4), b3)       ! Smoothness indicator of the first upwind stencil
+    else
+      call weno_seven_weight_0(q8(4:7), b0)
+      call weno_seven_weight_1(q8(3:6), b1)
+      call weno_seven_weight_2(q8(2:5), b2)
+      call weno_seven_weight_3(q8(1:4), b3)
+    endif
+  else
+    call weno_seven_reconstruction_0(q8(5:2:-1), c0) ! Reconstruction in the fourth upwind stencil
+    call weno_seven_reconstruction_1(q8(6:3:-1), c1) ! Reconstruction in the third upwind stencil
+    call weno_seven_reconstruction_2(q8(7:4:-1), c2) ! Reconstruction in the second upwind stencil
+    call weno_seven_reconstruction_3(q8(8:5:-1), c3) ! Reconstruction in the first upwind stencil
+
+    call weno_seven_reconstruction_0(h8(5:2:-1), d0)
+    call weno_seven_reconstruction_1(h8(6:3:-1), d1)
+    call weno_seven_reconstruction_2(h8(7:4:-1), d2)
+    call weno_seven_reconstruction_3(h8(8:5:-1), d3)
+    if (velocity_smoothing) then
+      call weno_seven_weight_0(u8(5:2:-1), b0)    ! Smoothness indicator of the fourth upwind stencil
+      call weno_seven_weight_1(u8(6:3:-1), b1)    ! Smoothness indicator of the third upwind stencil
+      call weno_seven_weight_2(u8(7:4:-1), b2)    ! Smoothness indicator of the second upwind stencil
+      call weno_seven_weight_3(u8(8:5:-1), b3)    ! Smoothness indicator of the first upwind stencil
+    else
+      call weno_seven_weight_0(q8(5:2:-1), b0)
+      call weno_seven_weight_1(q8(6:3:-1), b1)
+      call weno_seven_weight_2(q8(7:4:-1), b2)
+      call weno_seven_weight_3(q8(8:5:-1), b3)
+    endif
+  endif
+
+  tau = abs(b0 + 3 * b1 - 3 * b2 - b3)
+  w0  = 4./35.  * (1 + (tau / (b0 + 1e-20))**2)
```

Comment:
These expressions are dimensionally inconsistent. `b0` has arbitrary units that are the same as `tau`, but it can not be added to the nondimensional constant 1e-20.  Instead consider adding a tiny factor that scales with the range of `b0` through `b3` or perhaps a limit on huge the nondimesional weights `w0`, `w1`, `w2` and `w3` are allowed to be before they are normalized.

  Also add local parameter variables for the constants like 12./35., like `real, parameter :: C12_35 = 12.0/35.0 ! The ratio of 12 / 35 [nondim]`.